lease/lessor.go

```diff
@@ -160,9 +164,10 @@ func newLessor(b backend.Backend, minLeaseTTL int64) *lessor {
 		b:           b,
 		minLeaseTTL: minLeaseTTL,
 		// expiredC is a small buffered chan to avoid unnecessary blocking.
-		expiredC: make(chan []*Lease, 16),
-		stopC:    make(chan struct{}),
-		doneC:    make(chan struct{}),
+		expiredC:  make(chan []*Lease, 16),
+		stopC:     make(chan struct{}),
+		doneC:     make(chan struct{}),
+		promoting: true,
```

Comment:
this should be set to false initially? when promote is called, we should change promoting to true. after finishing promotion, we set it back to false again. right?